src/containers/CreateTaskRun/CreateTaskRun.js

```diff
@@ -223,29 +257,31 @@ class CreateTaskRun extends React.Component {
       validInputResources &&
       validOutputResources &&
       validParams &&
-      validTimeout &&
+      isValidTimeout &&
       validLabels &&
       validNodeSelector
     );
-  };
+  }
 
-  handleClose = () => {
-    const { defaultNamespace: namespace, history } = this.props;
-    const { kind, taskName } = this.getTaskDetails();
+  function handleClose() {
+    const { kind: taskKind, taskName } = getTaskDetails();
     let url = urls.taskRuns.all();
-    if (taskName && namespace !== ALL_NAMESPACES) {
-      url = urls.taskRuns[kind === 'ClusterTask' ? 'byClusterTask' : 'byTask']({
-        namespace,
+    if (taskName && defaultNamespace !== ALL_NAMESPACES) {
```

Comment:
I notice here that we are comparing `defaultNamespace` from props in the if statement, whereas in [CreatePipelineRun in the handleClose function](https://github.com/tektoncd/dashboard/pull/2037/files?diff=split&w=1#diff-a76903232a0bbce3293cb530f3aac343e2b5d6dda21048219bb2467ce9b5027dR255) we check against `namespace` from the state.

Should these match each other? The old code suggests we should be checking against `defaultNamespace` in both cases?

Reply:
nice catch, thanks. Fixed